### PR TITLE
Fix the bugs when sampling consistency models.

### PR DESCRIPTION
**Pull request by RossoneriZhao**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/diffusers/pipelines/pipeline_utils.py [hunk 1]
```diff
@@ -1066,27 +1077,51 @@ def load_module(name, value):
...
-                    offload_folder=offload_folder,
-                    offload_state_dict=offload_state_dict,
-                    model_variants=model_variants,
-                    name=name,
-                    from_flax=from_flax,
-                    variant=variant,
-                    low_cpu_mem_usage=low_cpu_mem_usage,
-                    cached_folder=cached_folder,
-                )
+                if name == "unet" and if_consistency:
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/pipeline_utils.py:1080]: We cannot do such a change in a core file such as pipeline_utils.py - any other way we could fix it?

Comment (RossoneriZhao) [reply src/diffusers/pipelines/pipeline_utils.py:1080]: Maybe deriving the class also works.

**Comment by sayakpaul** [comment]
@RossoneriZhao let us know if you need any help.

**Comment by sayakpaul** [comment]
@RossoneriZhao let us know if you need any help here.

**Comment by dg845** [comment]
**TL;DR**: Two suggestions for a fix that doesn't involve modifying `pipeline_utils.py`: **Suggestion 1** is to create consistency model-specific versions of `UNetMidBlock2D`/`UNet2DModel` that allows `resnet_time_scale_shift != default` and `Attention` in `UNetMidBlock2D` to have a group norm; **Suggestion 2** adds new arguments for `UNetMidBlock2D`/`UNet2DModel` to allow the default mid block `Attention` group norm behavior to be overridden. Also, what is the reasoning behind `resnet_time_scale_shift == default` and `Attention` having a group norm in `UNetMidBlock2D` being coupled?

----

**Suggestion 1**: Following @RossoneriZhao's suggestion in https://github.com/huggingface/diffusers/pull/4437#discussion_r1284095497, I think one possible (probably temporary) fix that doesn't involve directly changing `pipeline_utils.py` could be to to create a `modeling_consistency_models.py` file in `/diffusers/src/diffusers/pipelines/consistency_models/` and subclass `UNetMidBlock2D` and `UNet2DModel` to follow the consistency models implementation which allows both `resnet_time_scale_shift != "default"` and the `Attention` block's group norm to be present simultaneously, using code that is very similar to what's currently in the PR. We would then use this subclass of `UNet2DModel` in `ConsistencyModelPipeline`.

----

**Suggestion 2**: Another possible fix that might side step having to modify `pipeline_utils.py` is to create a new argument `attn_resnet_groups: Optional[int] = None` for `UNetMidBlock2D` which when set to `None` follows the current attention block group norm behavior:

```python
class UNetMidBlock2D(nn.Module):
    def __init__(...):
        ...
        if attn_resnet_groups is None:
            attn_resnet_groups = resnet_groups if resnet_time_scale_shift == "default" else None
        ...
```

and then when initializing the `Attention` block we would set `norm_num_groups=attn_resnet_groups`.

`UNet2DModel` would then have a corresponding new argument `attn_norm_num_groups: Optional[int] = None`, and when making the call to initialize `UNetMidBlock2D` we would add an additional argument `attn_resnet_groups=attn_norm_num_groups`.

Finally, to support the consistency models implementation, we would set `attn_norm_num_groups == norm_num_groups == 32` in the `UNet2DModel` config for consistency model checkpoints like [`openai/diffusers-cd_imagenet64_lpips`](https://huggingface.co/openai/diffusers-cd_imagenet64_lpips/blob/main/unet/config.json).

This is similar to the current implementation in the PR but should avoid having to modify `pipeline_utils.py` since existing `UNet2DModel` configs don't have a `attn_norm_num_groups` field, so it is set to `None` and thus the current mid block `Attention` group norm behavior is retained.

----

Finally, could someone provide some insight on why the `Attention` block has a group norm if and only if `resnet_time_scale_shift == default` in `UNetMidBlock2D`? As @RossoneriZhao mentioned earlier, it seems weird that these two things are coupled. @patrickvonplaten @sayakpaul

**Comment by dg845** [comment]
Hi @RossoneriZhao, are you still planning to work on this PR? If you're busy, I can work on completing this PR :) (likely following Suggestion 2 in https://github.com/huggingface/diffusers/pull/4437#issuecomment-1683210037, if that seems reasonable to people).

**Comment by dg845** [comment]
I have opened a PR based on Suggestion 2 above: #4863.
